src/compiler/cpp_generator.h

```diff
@@ -64,8 +64,15 @@ struct Parameters {
   grpc::string grpc_search_path;
 };
 
+// A common interface for objects having comments in the source.
+struct CommentHolder {
+  virtual ~CommentHolder() {}
+  virtual grpc::string GetLeadingComments() const = 0;
```

Comment:
Does this returned the comments that are already C++ formatted? If so, we might wanna change to name to indicate that or at least add an explanatory comment.